Plugin/Framework/Search/Request/GraphQlHandler.php

```diff
@@ -116,6 +122,10 @@ protected function preFetchOps(array $requestData)
     // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter
     public function parseLimit(array $requestData)
     {
+        if ($this->pageSize != -1) {
+            return $this->pageSize;
```

Comment:
Let's put a debug log when the DI value is used, this will make debugging easier in the future.

Reply:
Added log entry